File: launchdarkly/ruby/src/PostSegmentExample.rb
require "json"
require "launchdarkly_client"

LaunchDarklyClient.configure do |config|
    config.api_key["Authorization"] = "YOUR_API_KEY"
end

segment_body = LaunchDarklyClient::SegmentBody.new
segment_body.name = "Example segment"
segment_body.key = "segment-key-123abc"
segment_body.description = "Bundle our sample customers together"
segment_body.unbounded = false
segment_body.unbounded_context_kind = "device"
segment_body.tags = [
    "testing",
]

begin
    response = LaunchDarklyClient::SegmentsApi.new.post_segment(
        "projectKey_string", # project_key
        "environmentKey_string", # environment_key
        segment_body,
    )

    p response
rescue LaunchDarklyClient::ApiError => e
    puts "Exception when calling SegmentsApi#post_segment: #{e}"
end

File: launchdarkly/python/src/PostSegmentExample.py
import json
from datetime import date, datetime
from pprint import pprint

from launchdarkly_client import ApiClient, ApiException, Configuration, api, models

configuration = Configuration(
    api_key={"ApiKey": "YOUR_API_KEY"},
)

with ApiClient(configuration) as api_client:
    segment_body = models.SegmentBody(
        name="Example segment",
        key="segment-key-123abc",
        description="Bundle our sample customers together",
        unbounded=False,
        unboundedContextKind="device",
        tags=[
            "testing",
        ],
    )

    try:
        response = api.SegmentsApi(api_client).post_segment(
            project_key="projectKey_string",
            environment_key="environmentKey_string",
            segment_body=segment_body,
        )

        pprint(response)
    except ApiException as e:
        print("Exception when calling SegmentsApi#post_segment: %s\n" % e)
